### PR TITLE
Edge Cache i1: Edge cache feature flag tests

### PR DESCRIPTION
**Pull request by kozer**
Related to p1687766612113259-slack-C04GESRBWKW

## Proposed Changes

As a continuation of [this](https://github.com/Automattic/wp-calypso/pull/78622) fix, this pr adds some tests to test that everything works as expected regardless of the feature status.

## Testing Instructions

N/A

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/hosting/cache-card/test/index.js [hunk 1]
```diff
@@ -152,4 +155,49 @@ describe( 'CacheCard component', () => {
...
+		config.disable( EDGE_CACHE_FEATURE );
+		rerender(
+			<Provider store={ store }>
+				<CacheCard { ...{ ...defaultProps } } />
+			</Provider>
+		);
+		expect( screen.queryByRole( 'checkbox' ) ).toBeFalsy();
+	} );
+
+	it( 'shows the Clear Cache button regradless of the feature status', () => {
```
Comment (wojtekn) [inline client/my-sites/hosting/cache-card/test/index.js]:
```suggestion
	it( 'shows the Clear Cache button regardless of the feature status', () => {
```

**Review by wojtekn — COMMENTED** [review]
Thanks for adding those tests @kozer .

I'm uncertain whether these tests offer significant value. Would you mind clarifying their benefits? 

I've reversed changes from https://github.com/Automattic/wp-calypso/pull/78622 to add the bug again, executed tests, and they are still passing, meaning they would miss the issue we introduced.

**Comment by kozer** [comment]
> Thanks for adding those tests @kozer .
> 
> I'm uncertain whether these tests offer significant value. Would you mind clarifying their benefits?
> 
> I've reversed changes from #78622 to add the bug again, executed tests, and they are still passing, meaning they would miss the issue we introduced.

The first test is just there to test whether the checkbox is there, or not, given the feature status. It's not related to the bug, just a failsafe that everything works as expected.

For the second one, for me, it was failing if I remove the feature, but it wasn't 100% correct. Now it's fixed. Can you please give it a try?

**Comment by wojtekn** [comment]
> For the second one, for me, it was failing if I remove the feature, but it wasn't 100% correct. Now it's fixed. Can you please give it a try?

Thanks for checking it. I've checked it again with the previous version of the tests, and it works now. I think I might have applied the test changes incorrectly. Should we revert those changes to test for `toBeEnabled` in both cases?

**Comment by kozer** [comment]
> > For the second one, for me, it was failing if I remove the feature, but it wasn't 100% correct. Now it's fixed. Can you please give it a try?
> 
> Thanks for checking it. I've checked it again with the previous version of the tests, and it works now. I think I might have applied the test changes incorrectly. Should we revert those changes to test for `toBeEnabled` in both cases?

I think the changes I did now, capture better what it was going on:
- The `useEdgeCacheQuery` returns `isLoading` true, when the feature is disabled. This caused the button to be disabled.
This is tested now, and previously it wasn't captured as expected.
The `isEnabled` is tested implicitly by other tests, as we test `useEdgeCacheQuery` at some of them (this function couldn't be called if the button is disabled'

**Review by wojtekn — APPROVED** [review]

**Pull request merged**